sentry_sdk/integrations/trytond.py

```diff
@@ -0,0 +1,55 @@
+import sentry_sdk.hub
+import sentry_sdk.utils
+import sentry_sdk.integrations
+import sentry_sdk.integrations.wsgi
+from sentry_sdk._types import MYPY
+
+from trytond.exceptions import TrytonException  # type: ignore
+from trytond.wsgi import app  # type: ignore
+
+if MYPY:
+    from typing import Any
+
+
+# TODO: trytond-worker, trytond-cron and trytond-admin intergations
+
+
+class TrytondWSGIIntegration(sentry_sdk.integrations.Integration):
+    identifier = "trytond_wsgi"
+
+    def __init__(self):  # type: () -> None
+        pass
+
+    @staticmethod
+    def setup_once():  # type: () -> None
+
+        app.wsgi_app = sentry_sdk.integrations.wsgi.SentryWsgiMiddleware(app.wsgi_app)
+
+        def error_handler(e):  # type: (Exception) -> None
+            hub = sentry_sdk.hub.Hub.current
+
+            if hub.get_integration(TrytondWSGIIntegration) is None:
+                return
+            elif isinstance(e, TrytonException):
+                return
+            else:
+                # If an integration is there, a client has to be there.
+                client = hub.client  # type: Any
+                event, hint = sentry_sdk.utils.event_from_exception(
+                    e,
```

Comment:
Switched back to build the event from the caught exception instead of sys.exc_info() because it feels more elegant to me. Does using sys.exc_info() have any advantatge I am missing?

Reply:
Under Python 2 this will cause you to lose the traceback, but it should be fine in this context.